pkg/patterns/declarative/pkg/applier/direct.go

```diff
@@ -27,24 +26,41 @@ func (d *DirectApplier) Apply(ctx context.Context,
 	validate bool,
 	extraArgs ...string,
 ) error {
+
+	tmpFile, err := ioutil.TempFile("", "tmp-manifest-*.yaml")
+	if err != nil {
+		return err
+	}
+	tmpFile.Write([]byte(manifest))
```

Comment:
Nit: the error handling here is really tricky.  I would split this into a function, like this:

```
func WriteTempFile(dir, pattern string, contents []byte) (string, error) {
  tmpFile, err := ioutil.TempFile(dir, pattern)
  if err != nil {
    return "", err
  }
  success := false
  defer func() {
    if !success {
      tmpFile.Close()
      os.Remove(tmpFile.Name())
    }
  }()
  if _, err :=  tmpFile.Write(contents); err != nil {
    return "", err
  }
  if err := tmpFile.Close(); err != nil {
    return "", err
  }
  success = true // don't remove the temp file
  return tmpFile.Name(), nil
}
```


I'm happy to send a follow-on PR to add this if that's easier!

Reply:
Thank you to comment :smile: 
Current codes are something rough idea, so any review comments are very appreciate.